### ISSUE TITLE
Bootstrap collapse close animation breaks when builiding in Angular's production mode

### ISSUE THREAD

**Issue by qbus00** [twbs/bootstrap#35776]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I don't know if this can be treated as a bug of Bootstrap or just a specific usage of Javascript but I think it's worth to mention it here.

There is an issue with close animation of collapse panel when running Angular project with Bootstrap in production mode.
You can check it here: https://github.com/qbus00/angular-bootstrap-collapse-test

I have reported a bug to Angular-CLI team - https://github.com/angular/angular-cli/issues/22635 - but they have closed it describing it as "by design".

Quote from reply (can be also found under aforementioned link:

> Hi, I investigated this morning and this seems to be caused by [pure_getters](https://github.com/terser/terser#compress-options) terser option. Bootstrap relies on side effects example https://github.com/twbs/bootstrap/blob/a5483a8a96eaa88b4da7e432e0b925d0f3943dc3/js/src/collapse.js#L182 therefore in this case it is expected that the code breaks during runtime.
> There are a couple of resolutions for this that you can read more about in https://github.com/angular/angular-cli/issues/11439#issuecomment-433141561
> Closing as this is working as intended.

As I understand it's because Bootstrap's JS is using getters not only to access property but also modifies something, which when running minification breaks the functionality.

Is there something you can do about that? Maybe rewrite this portion of Javascript - instead of getters maybe just a simple function would fix this?

Mixture of Angular and Bootstrap is not something uncommon....

### Reduced test cases

1. Clone: https://github.com/qbus00/angular-bootstrap-collapse-test
2. "npm install"
3. "ng serve --configuration production"
4. run browser
5. click on CLICK ME button
6. expand animation works
7. click again CLICK ME button
8. collapsing animation does not work
9. stop ng serve
10. repeat steps 3 to 9 but instead of "ng serve --configuration production" run "ng serve"
11. collapsing animation now works (no optimization of bootstrap.js file)

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### What browser(s) are you seeing the problem on?

Chrome, Firefox, Microsoft Edge, Opera

### What version of Bootstrap are you using?

5.1, 5.0

**Comment by mdo** [comment]
@twbs/js-review Any thoughts?

**Comment by rarb1t** [comment]
Hi qbus00! I think something I tried earlier solves this issue.  I'm new to Angular and everything related to web development, so I'm aware this might not be the best approach but hey! It solves the animation problem after all.

So I first read your bug report and saw that the recommended "solution" was to opt-out from the uglify optimizations implemented by Angular. I tried that but the size of my project exceeded the 1MB budget, so I ended up not doing it and I removed the line disabling script optimizations from my angular.json file.

However by checking the terminal outputs where the raw size of the /dist files are shown, I noticed that scripts.js did not increase or reduce it's size while uglify was turned off. This made me realize that bootstrap.min.js was not taking advantage of any optimizations.

I then thought: _**"What if I copy the original Bootstrap JS file from node_modules, rename it to what the builder calls it and replace the outputted file inside /dist?"**_

Well, I did that and it looks like it worked! I deployed my app to GitHub Pages making sure it's not rebuilt and the collapser is working as it should. You can see the solution in action [here](https://rarb1t.github.io/angular-app/) (collapser visible for small screens only).

Other than removing the commented lines from the original bootstrap.min.js file, I did not change anything else. The app is based on [this](https://www.youtube.com/watch?v=G0bBLvWXBvc&ab_channel=Fireship) tutorial by Fireship if anyone is wondering, great guide!

Anyways, I hope this workaround helps you and everyone else struggling with this small problem that yet was haunting me daily.

**Comment by rarb1t** [comment]
Yes, you are correct. I changed my wording and have updated my comment, glad to know you also got it to work

**Comment by XhmikosR** [comment]
@GeoSot @julien-deramond is this fixed easily without breaking anything else? If so, could we land it in v5.2.2 maybe?

**Comment by XhmikosR** [comment]
Otherwise, if we can't fix it in the source code, we might as well disable the related terser option?

**Comment by urmik-zwt** [comment]
Having the same problem in my Angular14 project after the production build. So I have added temporary "configurations":{"optimization": false} in the angular.json file so it is working fine for me.

I am new to angular so I am not sure if this is valid or not.

**Comment by XhmikosR** [comment]
It's valid, but you are totally disabling production optimizations altogether which is bad for production.

I'm still not sure what we can do to fix this.

**Comment by mohamed-badaoui** [comment]
Still having the issue with Angular 15.2.6 and Bootstrap 5.2.

Animation works fine in dev environment
![2023-04-09-dev-env](https://user-images.githubusercontent.com/3098863/230788081-cc96e73e-e589-427a-a175-1916b51147b9.gif)

But in prod environment, collapse close animation is broken
![2023-04-09-prod-env](https://user-images.githubusercontent.com/3098863/230788085-ed6a856d-0bb5-4941-9a68-2325a6c53e3a.gif)

**Comment by XhmikosR** [comment]
Unsure how we can work around this? If anyone has any non-breaking suggestions, please make a PR and CC me.

**Comment by harrisonwlee** [comment]
Has there been any update on this issue yet?

**Comment by gayatrisuslade26** [comment]
I was having the same problem in my Angular16 project after the production build. So I have added "configurations":{"optimization": false} in the angular.json for production mode in the  file then it is working fine for me.

**Comment by XhmikosR** [comment]
I'll say it again in case people miss it: https://github.com/twbs/bootstrap/issues/35776#issuecomment-1312736536

I'd be more than happy to accept a PR that fixes the issue, ideally accompanied with tests.

**Comment by pablovargasosorio** [comment]
Having the same issue with Angular v17.3.0 / Bootstrap v5.3.3
It's unbelievable this is failing and not fixed yet by Angular or Bootstrap teams. Disabling 'optimization' for a project in Production only for this silly thing makes me angry to be honest...

There's another alternative workaround, which is including the bootstrap source from the index.html as described here: https://stackoverflow.com/questions/70159067/angular-bootstrap-accordion-animation-is-not-working-on-close-in-production

It makes me angry too...

**Comment by pablovargasosorio** [comment]
This worked for me: **ngbAccordion** works perfectly with 'optimization'.
It's easy to change from 'accordion' to 'ngbAccordion' in the HTML (if you have ngBootstrap installed in your project)

More info: https://ng-bootstrap.github.io/#/components/accordion/overview
